src/app/modules/submission/email-submission/email-submission.controller.ts

```diff
@@ -30,267 +32,274 @@ const logger = createLoggerWithLabel(module)
+              message: 'Error while saving metadata to database',
+              meta: logMeta,
+              error,
+            })
+            return error
+          })
+      })
+      .andThen(({ form, parsedResponses, submission, emailData }) => {
+        const logMetaWithSubmission = {
+          ...logMeta,
+          submissionId: submission._id,
+        }
 
-  // MyInfo access token is single-use, so clear it
-  res.clearCookie(MYINFO_COOKIE_NAME, MYINFO_COOKIE_OPTIONS)
-  // Return the reply early to the submitter
-  res.json({
-    message: 'Form submission successful.',
-    submissionId: submission.id,
-  })
+        logger.info({
+          message: 'Sending admin mail',
+          meta: logMetaWithSubmission,
+        })
 
-  // Send email confirmations
-  return SubmissionService.sendEmailConfirmations({
-    form,
-    parsedResponses,
-    submission,
-    attachments,
-    autoReplyData: emailData.autoReplyData,
-  }).mapErr((error) => {
-    logger.error({
-      message: 'Error while sending email confirmations',
-      meta: logMetaWithSubmission,
-      error,
-    })
-  })
+        // Send response to admin
+        // NOTE: This should short circuit in the event of an error.
+        // This is why sendSubmissionToAdmin is separated from sendEmailConfirmations in 2 blocks
+        return MailService.sendSubmissionToAdmin({
+          replyToEmails: EmailSubmissionService.extractEmailAnswers(
+            parsedResponses,
+          ),
+          form,
+          submission,
+          attachments,
+          dataCollationData: emailData.dataCollationData,
+          formData: emailData.formData,
+        })
+          .map(() => ({
+            form,
+            parsedResponses,
+            submission,
+            emailData,
+            logMeta,
+          }))
+          .mapErr((error) => {
+            logger.error({
+              message: 'Error sending submission to admin',
+              meta: logMetaWithSubmission,
+              error,
+            })
+            return error
+          })
+      })
+      .map(({ form, parsedResponses, submission, emailData, logMeta }) => {
+        // Send email confirmations
+        void SubmissionService.sendEmailConfirmations({
+          form,
+          parsedResponses,
+          submission,
+          attachments,
+          autoReplyData: emailData.autoReplyData,
+        }).mapErr((error) => {
+          logger.error({
+            message: 'Error while sending email confirmations',
+            meta: logMeta,
+            error,
+          })
+        })
+        // MyInfo access token is single-use, so clear it
+        return res.clearCookie(MYINFO_COOKIE_NAME, MYINFO_COOKIE_OPTIONS).json({
+          // Return the reply early to the submitter
+          message: 'Form submission successful.',
+          submissionId: submission.id,
+        })
+      })
+      .mapErr((error) => {
+        const { errorMessage, statusCode } = mapRouteError(error)
```

Comment:
This is not in the original code; but since myinfo token is cleared in the success case, we should also clear it in the error case right? 

double check my assertion @mantariksh

Reply:
in the original code, we clear the cookie regardless once the email submission is successfully sent isn't it?

Reply:
Yea, i meant we should also clear the token in the error case; and this clearing of the token in the error case is not in the original code.

Reply:
no, we should not clear it in the error case. if the submission fails through no fault of the user (eg they submitted before captcha could connect, or the admin email failed to send), they should not be randomly signed out.

Reply:
I see, thanks for the explanation! Should we add it in the comments too?

Reply:
thanks for explaining; will add this in comments for future devs!